sklearn/feature_extraction/tests/test_feature_hasher.py

```diff
@@ -53,6 +54,17 @@ def test_feature_hasher_pairs():
     assert_equal([1, 3, 4], x2_nz)
 
 
+def test_feature_hasher_pairs_with_string_values():
+    raw_X = (iter(d.items()) for d in [{"foo": 1, "bar": "a"},
+                                       {"baz": u"abc", "quux": 4, "foo": -1}])
+    h = FeatureHasher(n_features=16, input_type="pair")
+    x1, x2 = h.transform(raw_X).toarray()
+    x1_nz = sorted(np.abs(x1[x1 != 0]))
+    x2_nz = sorted(np.abs(x2[x2 != 0]))
+    assert_equal([1, 1], x1_nz)
```

Comment:
when we'll transform the first dict (stored in x1), the "a" will be hashed as 1. Here `x1` will be: `[ 0.  0.  1.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0.  1.  0.  0.]` and `x2` will be: `[ 0.  0.  0.  1.  4.  0.  0.  0.  0.  0.  0.  0.  0.  0.  0. -1.]`. This test is just to show that indeed the string value is being hashed as 1.